mitodl/mit-open
# Removes generated API client from repo, replacing with package import

### What are the relevant tickets?





Technical debt.

### Description (What does it do?)


Removes the generated API client from the repository, replaced by importing `@mitodl/open-api-axios` from NPM.

Re-exports v1 and v2 for type imports.


### How can this be tested?

The `main` workspace should build and run successfully after installing dependencies.

```
yarn install
yarn workspace main build
yarn workspace main dev
docker compose up
```

### Additional Context



The [API client](https://github.com/mitodl/open-api-clients) distribution is [published to NPM](https://www.npmjs.com/package/@mitodl/open-api-axios), though also committed to this repo at the generated client inside the api workspace, ./frontends/api/src/generated.

This is not necessary as it can be imported normally as a dependency. If we do not pin the dependency (use a ranged ^ version) we discourage breaking API changes within our existing v1 and v2 APIs.
